xla/service/scatter_utils.cc

```diff
@@ -0,0 +1,217 @@
+
+#include "xla/service/scatter_utils.h"
+
+#include "xla/service/hlo_creation_utils.h"
+#include "xla/service/call_inliner.h"
+
+namespace xla {
+
+StatusOr<HloInstruction*> TransposeIndexVectorDimToLast(
+    HloInstruction* scatter_indices, int64_t index_vector_dim) {
+  const Shape& scatter_indices_shape = scatter_indices->shape();
+
+  if (scatter_indices_shape.dimensions_size() == index_vector_dim) {
+    return scatter_indices;
+  }
+
+  if (index_vector_dim == (scatter_indices_shape.dimensions_size() - 1)) {
```

Comment:
I think the if above and the if here can be simplified into one if block to:

```
if (index_vector_dim >= scatter_indices_shape.dimensions_size() - 1) {
```

Reply:
Changed as requested